packages/sanity/src/core/releases/tool/components/ReleaseMenuButton/ReleaseMenuButton.tsx

```diff
@@ -19,35 +20,117 @@ export type ReleaseMenuButtonProps = {
 const ARCHIVABLE_STATES = ['active', 'published']
 
 export const ReleaseMenuButton = ({disabled, release}: ReleaseMenuButtonProps) => {
-  const {archive, unarchive} = useReleaseOperations()
+  const toast = useToast()
+  const {archive} = useReleaseOperations()
+  const {loading: isLoadingReleaseDocuments, results: releaseDocuments} = useBundleDocuments(
+    getBundleIdFromReleaseDocumentId(release._id),
+  )
   const [isPerformingOperation, setIsPerformingOperation] = useState(false)
   const [selectedAction, setSelectedAction] = useState<'edit' | 'confirm-archive'>()
 
-  const releaseMenuDisabled = !release || disabled
+  const releaseMenuDisabled = !release || isLoadingReleaseDocuments || disabled
   const {t} = useTranslation(releasesLocaleNamespace)
   const telemetry = useTelemetry()
 
-  const handleArchive = async (e: Parameters<FormEventHandler<HTMLFormElement>>[0]) => {
+  const handleArchive = useCallback(async () => {
     if (releaseMenuDisabled) return
-    e.preventDefault()
 
-    setIsPerformingOperation(true)
-    await archive(release._id)
+    try {
+      setIsPerformingOperation(true)
+      await archive(release._id)
 
-    // it's in the process of becoming true, so the event we want to track is archive
-    telemetry.log(ArchivedRelease)
-    setIsPerformingOperation(false)
-  }
+      // it's in the process of becoming true, so the event we want to track is archive
+      telemetry.log(ArchivedRelease)
+      toast.push({
+        closable: true,
+        status: 'success',
+        title: (
+          <Text muted size={1}>
+            <Translate
+              t={t}
+              i18nKey="toast.archive.success"
+              values={{title: release.metadata.title}}
+            />
+          </Text>
+        ),
+      })
+    } catch (archivingError) {
+      toast.push({
+        status: 'error',
+        title: (
+          <Text muted size={1}>
+            <Translate
+              t={t}
+              i18nKey="toast.archive.error"
+              values={{title: release.metadata.title, error: archivingError.toString()}}
+            />
+          </Text>
+        ),
+      })
+      console.error(archivingError)
+    } finally {
+      setIsPerformingOperation(false)
+      setSelectedAction(undefined)
+    }
+  }, [archive, release._id, release.metadata.title, releaseMenuDisabled, t, telemetry, toast])
 
   const handleUnarchive = async () => {
-    setIsPerformingOperation(true)
-    await unarchive(release._id)
-
-    // it's in the process of becoming false, so the event we want to track is unarchive
-    telemetry.log(UnarchivedRelease)
-    setIsPerformingOperation(false)
+    // noop
+    // TODO: similar to handleArchive - complete once server action exists
   }
 
+  const confirmArchiveDialog = useMemo(() => {
+    if (selectedAction !== 'confirm-archive') return null
+
+    const dialogDescription =
+      releaseDocuments.length === 1
```

Comment:
Could we use the _one, _other syntax for these? https://www.i18next.com/translation-function/plurals#languages-with-multiple-plurals

Reply:
Ah, I wasn't a massive fan of `other` since it isn't really clear it means 'not one', but I see it's used in many placed, and also proposed by i18next, so yeh I'll capitulate and change to it